src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -404,24 +405,56 @@ def _handleNoStatusUpdate(self, workload, request_args, dn):
         """
         For no-status update, we only support the following parameters:
          1. RequestPriority
-         2. Global workqueue statistics, while acquiring a workflow
+         2. SiteWhitelist
+         3. SiteBlacklist
+         4. Global workqueue statistics, while acquiring a workflow
+        As Global workqueue statistics updates are exclusive to the rest of the
+        parameters. Meaning if it is about to be updated all the rest of the
+        request_args will be ignored.
         """
-        if 'RequestPriority' in request_args:
-            # Yes, we completely ignore any other arguments posted by the user (web UI case)
-            request_args = {'RequestPriority': request_args['RequestPriority']}
-            validate_request_priority(request_args)
-            # must update three places: GQ elements, workload_cache and workload spec
-            self.gq_service.updatePriority(workload.name(), request_args['RequestPriority'])
-            report = self.reqmgr_db_service.updateRequestProperty(workload.name(), request_args, dn)
-            workload.setPriority(request_args['RequestPriority'])
-            workload.saveCouchUrl(workload.specUrl())
-            cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), request_args['RequestPriority']))
-        elif workqueue_stat_validation(request_args):
-            report = self.reqmgr_db_service.updateRequestStats(workload.name(), request_args)
-            cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), request_args))
-        else:
-            msg = "There are invalid arguments for no-status update: %s" % request_args
-            raise InvalidSpecParameterValue(msg)
+        reqArgs = deepcopy(request_args)
```

Comment:
We have already discussed this line in your other PR, but can you please remind me why we need to have a full copy of the object at this place? 
From what I can tell, we only pop `RequestName` in the method `validate_request_update_args()`, which is called here: https://github.com/dmwm/WMCore/blob/62a6d43ecd268b2fb2154070ba801bd61d4a87a3/src/python/WMCore/ReqMgr/Service/Request.py#L172

So making this is a tardy use of deepcopy, no?

Reply:
because the whole chain of calls for validating and implementing a change to the  request is done through destructive methods, this one not making any difference. And some of the logic (also some of it was mentioned by you yourself during one of those meetings) was relying on those destructive calls during the validation etc. etc.  Hence we'd like to act here only through a completely separate copy of the `request_arguments` (which is really nothing big in terms of size) and not alter the already existing logic of validating the request arguments. So in short - to stay on the safe side.